specs/_features/execution_proofs/beacon-chain.md

```diff
@@ -0,0 +1,214 @@
+
+## Table of contents
+
+<!-- mdformat-toc start --slug=github --no-anchors --maxlevel=6 --minlevel=1 -->
+
+- [Execution Proofs -- The Beacon Chain](#execution-proofs----the-beacon-chain)
+  - [Table of contents](#table-of-contents)
+  - [Introduction](#introduction)
+  - [Constants](#constants)
+    - [Execution](#execution)
+  - [Configuration](#configuration)
+  - [Containers](#containers)
+    - [New containers](#new-containers)
+      - [`ExecutionProof`](#executionproof)
+      - [`SignedExecutionProof`](#signedexecutionproof)
+    - [Extended Containers](#extended-containers)
+  - [Helper functions](#helper-functions)
+    - [Execution proof functions](#execution-proof-functions)
+      - [`get_el_program`](#get_el_program)
+      - [`verify_execution_proof`](#verify_execution_proof)
+      - [`verify_execution_proofs`](#verify_execution_proofs)
+  - [Beacon chain state transition function](#beacon-chain-state-transition-function)
+    - [Execution payload processing](#execution-payload-processing)
+      - [Modified `process_execution_payload`](#modified-process_execution_payload)
+
+<!-- mdformat-toc end -->
+
+## Introduction
+
+This document contains the consensus specs for Execution Proofs. This enables stateless validation of execution payloads through cryptographic proofs.
+
+*Note*: This specification assumes the reader is familiar with the [zkEVM cryptographic operations](./zkevm.md).
+
+## Constants
+
+### Execution
+
+| Name                                    | Value             |
+| --------------------------------------- | ----------------- |
+| `MAX_EXECUTION_PROOFS_PER_PAYLOAD`     | `uint64(8)`       |
+| `RETH_EL_PROGRAM`                      | `EL_PROGRAM(b"RETH_V1" + b"\x00" * 25)` |
+| `GETH_EL_PROGRAM`                      | `EL_PROGRAM(b"GETH_V1" + b"\x00" * 25)` |
+| `NETHERMIND_EL_PROGRAM`                | `EL_PROGRAM(b"NETHERMIND_V1" + b"\x00" * 20)` |
+| `BESU_EL_PROGRAM`                      | `EL_PROGRAM(b"BESU_V1" + b"\x00" * 25)` |
+
+## Configuration
+
+*Note*: The configuration values are not definitive.
+
+| Name                             | Value                           |
+| -------------------------------- | ------------------------------- |
+| `MIN_REQUIRED_EXECUTION_PROOFS`  | `uint64(1)`                     |
+
+## Containers
+
+### New containers
+
+#### `ExecutionProof`
+
+```python
+class ExecutionProof(Container):
+    beacon_root: Root
+    zk_proof: ZKProof
+    validator_index: ValidatorIndex
+```
+
+#### `SignedExecutionProof`
+
+```python
+class SignedExecutionProof(Container):
+    message: ExecutionProof
+    signature: BLSSignature
+```
+
+### Extended Containers
+
+*Note*: `BeaconState` and `BeaconBlockBody` remain the same as Fulu. No modifications are required for execution proofs since they are handled externally.
+
+## Helper functions
+
+### Execution proof functions
+
+#### `get_el_program`
+
+```python
+def get_el_program(proof_id: ProofID) -> EL_PROGRAM:
```

Comment:
This is a somehwta weird function that we could put in the zkevm.md file -- given zkevm already implies knowledge of an EL

Reply:
removed since we no longer have multiple clients